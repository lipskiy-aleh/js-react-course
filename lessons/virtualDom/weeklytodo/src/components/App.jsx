import { WeeklyTodo } from './WeeklyTodo'

export function App() {
  return (
    <div className="App">
      <WeeklyTodo />
    </div>
  );
}
