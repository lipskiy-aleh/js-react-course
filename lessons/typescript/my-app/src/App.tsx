import React from 'react';
import { Hello } from './components/Hello'
import { Counter } from './components/Counter'

function App() {
  return (
    <div>
      <Hello name="Test"/>
      <Counter />
    </div>
  );
}

export default App;
