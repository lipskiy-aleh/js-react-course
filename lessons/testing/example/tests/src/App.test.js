import React from 'react'
import App from './App';
import { shallow, mount } from 'enzyme';


describe('App', () => {
  it('shallow', () => {
    const cmp = shallow(<App />)

    expect(cmp).toMatchSnapshot()
  })

  it('mount', () => {
    const cmp = mount(<App />)

    const link = cmp.find('a')
    expect(link.prop('href')).toBe('https://reactjs.org')
    expect(cmp).toMatchSnapshot()
  })
})

// test('renders learn react link', () => {
//   const cmp = shallow(<App />)

//   const link = cmp.find('a')
//   // link.prop()
//   // link.props()

//   expect(cmp).toMatchSnapshot()
//   // expect(link.length).toBe(1)
//   // expect(link.prop('href')).toBe('https://reactjs.org')
//   // expect(link.text()).toBe('Learn React')
//   // expect(link.props()).toEqual({})
// });

