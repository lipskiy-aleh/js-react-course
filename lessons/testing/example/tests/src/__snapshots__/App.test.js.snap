// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App mount 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <Link
        href="https://reactjs.org"
        text="App-link"
      >
        <a
          className="App-link"
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          App-link
        </a>
      </Link>
    </header>
  </div>
</App>
`;

exports[`App shallow 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <Link
      href="https://reactjs.org"
      text="App-link"
    />
  </header>
</div>
`;
